tools/wavetool.py

```diff
@@ -0,0 +1,157 @@
+    split_cmd_params = cmd_param.split(';')
+    wave_params = [param.strip() for param in split_cmd_params]
+    param_dict = dict(zip(param_keys, wave_params))
+    if param_dict['type'] not in supported_type:
+        print("Unsupported wave type")
+        sys.exit(1)
+    if param_dict['type'] in ['sine', 'cosine']:
+        wave_data = generateSinusoid(**param_dict)
+        return wave_data, param_dict
+
+def do_wave_compare(cmd_args):
+    supported_mode = ['HWEchoRef']
+    cmd_line = cmd_args['compare'].strip().split(';')
+    comparison_mode = cmd_line[0].strip()
+    if comparison_mode not in supported_mode:
+        print("Unsupported comparison mode: %s" % comparison_mode)
+        print('Supported comparison modes are: ' + str(supported_mode))
+        sys.exit(1)
+    wav_path, ref_path = [path.strip() for path in cmd_line[1:]]
+    if not os.path.exists(wav_path):
+        print('Recorded wave path does not exist: %s' % wav_path)
+        sys.exit(1)
+    if not os.path.exists(ref_path):
+        print('Reference wave path does not exist: %s' % ref_path)
+        sys.exit(1)
+    if comparison_mode == 'HWEchoRef':
+        compare_wav_hw_echo_ref(wav_path, ref_path)
+
+def save_wave(wave_data, path, **wave_params):
+    supported_format = ['S8', 'S16', 'S32', 'F32']
+    wave_path = path
+    if path == '.' or not path.endswith('wav'):
+        wave_name = wave_params['type'] + '_' + wave_params['channel'] + 'ch_' + wave_params['freq'] + 'Hz_' \
+        + wave_params['sample_rate'] + '_' + wave_params['format'] + '.wav'
+        wave_path = path + '/'  + wave_name
+    wave_format = wave_params['format'].upper()
+    if wave_format not in supported_format:
+        print('Unsupported wave format: %s' % wave_format)
+        sys.exit(1)
+    if wave_format == 'S8':
+        wave_data = (np.iinfo(np.int8).max * wave_data).astype(np.int8)
+    if wave_format == 'S16':
+        wave_data = (np.iinfo(np.int16).max * wave_data).astype(np.int16)
+    if wave_format == 'S32':
+        wave_data = (np.iinfo(np.int32).max * wave_data).astype(np.int32)
+    try:
+        wavefile.write(wave_path, int(wave_params['sample_rate']), wave_data)
+    except:
+        print("Path specified not valid: %s" % wave_path)
+
+# remove zeros in two sides
+def trim_wave(wave):
+    wave_mono = wave[:,0]
+    left_idx = 0
+    right_idx = wave_mono.shape[0] - 1
+    while True:
+        if wave_mono[left_idx] != 0:
+            break
+        left_idx = left_idx + 1
+    while True:
+        if wave_mono[right_idx] != 0:
+            break
+        right_idx = right_idx - 1
+    return wave[left_idx:right_idx,:], left_idx
+
+# float point binary comparison is not supported, and will not be supported
+# check HW echo referece wave
+def compare_wav_hw_echo_ref(wave_path, ref_path):
+    fs_wav, wave = wavefile.read(wave_path)
+    fs_ref, ref_wav = wavefile.read(ref_path)
```

Comment:
reading the files outside this function would probably make it easier to re-use.

Reply:
thanks, will do